pkg/transformer/openshift/openshift.go

```diff
@@ -235,5 +240,73 @@ func (o *OpenShift) Deploy(komposeObject kobject.KomposeObject, opt kobject.Conv
 }
 
 func (o *OpenShift) Undeploy(komposeObject kobject.KomposeObject, opt kobject.ConvertOptions) error {
-	return errors.New("Not Implemented")
+	//Convert komposeObject
+	objects := o.Transform(komposeObject, opt)
+
+	// initialize OpenShift Client
+	loadingRules := ocliconfig.NewOpenShiftClientConfigLoadingRules()
+	overrides := &clientcmd.ConfigOverrides{}
+	oclientConfig, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, overrides).ClientConfig()
+	if err != nil {
+		return err
+	}
+	oclient := oclient.NewOrDie(oclientConfig)
+
+
+	// initialize Kubernetes client
+	kfactory := kcmdutil.NewFactory(nil)
+	kclientConfig, err := kfactory.ClientConfig()
+	if err != nil {
+		return err
+	}
+	kclient := kclient.NewOrDie(kclientConfig)
+
+	// get namespace from config
+	namespace, _, err := kfactory.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	for _, v := range objects {
+		switch t := v.(type) {
+		case *imageapi.ImageStream:
+			//delete imageStream
+			err = oclient.ImageStreams(namespace).Delete(t.Name)
+			if  err != nil {
+				return err
+			} else {
+				logrus.Infof("Successfully deleted ImageStream: %s", t.Name)
+			}
+		case *deployapi.DeploymentConfig:
+			// delete deploymentConfig
+			err = oclient.DeploymentConfigs(namespace).Delete(t.Name)
```

Comment:
Using reaper function [`Stop`](https://github.com/openshift/origin/blob/master/pkg/deploy/cmd/delete.go#L44) would be better, than just deleting it right away.

I tried something like this 

```diff
diff --git a/pkg/transformer/openshift/openshift.go b/pkg/transformer/openshift/openshift.go
index 8a81fef..992c364 100644
--- a/pkg/transformer/openshift/openshift.go
+++ b/pkg/transformer/openshift/openshift.go
@@ -35,11 +35,13 @@ import (
 
 	oclient "github.com/openshift/origin/pkg/client"
 	ocliconfig "github.com/openshift/origin/pkg/cmd/cli/config"
+	deploymentconfigreaper "github.com/openshift/origin/pkg/deploy/cmd"
+
+	"time"
 
 	deployapi "github.com/openshift/origin/pkg/deploy/api"
 	imageapi "github.com/openshift/origin/pkg/image/api"
 	"k8s.io/kubernetes/pkg/kubectl"
-	"time"
 )
 
 type OpenShift struct {
@@ -252,7 +254,6 @@ func (o *OpenShift) Undeploy(komposeObject kobject.KomposeObject, opt kobject.Co
 	}
 	oclient := oclient.NewOrDie(oclientConfig)
 
-
 	// initialize Kubernetes client
 	kfactory := kcmdutil.NewFactory(nil)
 	kclientConfig, err := kfactory.ClientConfig()
@@ -272,14 +273,16 @@ func (o *OpenShift) Undeploy(komposeObject kobject.KomposeObject, opt kobject.Co
 		case *imageapi.ImageStream:
 			//delete imageStream
 			err = oclient.ImageStreams(namespace).Delete(t.Name)
-			if  err != nil {
+			if err != nil {
 				return err
 			} else {
 				logrus.Infof("Successfully deleted ImageStream: %s", t.Name)
 			}
 		case *deployapi.DeploymentConfig:
 			// delete deploymentConfig
-			err = oclient.DeploymentConfigs(namespace).Delete(t.Name)
+			dcreaper := deploymentconfigreaper.NewDeploymentConfigReaper(oclient, kclient)
+			err := dcreaper.Stop(namespace, t.Name, TIMEOUT*time.Second, nil)
+			//err = oclient.DeploymentConfigs(namespace).Delete(t.Name)
 			if err != nil {
 				return err
 			} else {
```